talk/tools.tex

```diff
@@ -683,3 +683,49 @@
     \end{itemize}
   \end{alertblock}
 \end{frame}
+
+\begin{frame}[fragile]
+  \frametitle{clang-tidy}
+  \begin{block}{compile\_commands.json}
+    \begin{itemize}
+      \item clang-tidy needs to know exactly how your program is built
+      \item tries to read this from a file called compile\_commands.json
+      \item generate with CMake: \\
+        cmake -DCMAKE\_EXPORT\_COMPILE\_COMMANDS=ON ...
+      \item for Makefiles try \href{https://github.com/rizsotto/Bear}{\beamergotobutton{Bear}}
+    \end{itemize}
+  \end{block}
+  \begin{block}{.clang-tidy}
+    \begin{itemize}
+      \item file describing which checks to run
+    \end{itemize}
+  \end{block}
+  \begin{block}{Supported checks}
+    \begin{itemize}
+      \item \url{https://clang.llvm.org/extra/clang-tidy/checks/list.html}
+    \end{itemize}
+  \end{block}
+\end{frame}
+
+\begin{frame}[fragile]
+  \frametitle{clang-tidy}
+  \begin{block}{Run clang-tidy}
+    \begin{itemize}
+      \item clang-tidy \textless{}file.cpp\textgreater{} -checks=...
+      \item clang-tidy \textless{}file.cpp\textgreater{} (checks from .clang-tidy tile)
+      \item clang-tidy \textless{}file.cpp\textgreater{} -{}-fix (applies fixes automatically)
+      \item run-clang-tidy (all files, checks from .clang-tidy)
+      \item run-clang-tidy -fix (applies fixes automatically)
+    \end{itemize}
+  \end{block}
+  \begin{alertblock}{Exercise Time}
+    \begin{itemize}
+      \item go to any example
+      \item cmake -DCMAKE\_EXPORT\_COMPILE\_COMMANDS=ON .
```

Comment:
```suggestion
      \item \mintinline{bash}{cmake -DCMAKE\_EXPORT\_COMPILE\_COMMANDS=ON .}
```
The last line leaks out of the slide. Maybe it fits if this one is typeset as shell code?

Reply:
I was also wondering whether you use a different typesetting environment for bash commands, but I did not look around far enough. Thx!